app/models/ta.rb

```diff
@@ -50,24 +72,48 @@ def get_criterion_associations_by_assignment(assignment)
 
   def get_criterion_associations_count_by_assignment(assignment)
     assignment.criterion_ta_associations
-              .where(ta_id: id)
-              .count
+      .where(ta_id: id)
+      .count
   end
 
   def get_membership_count_by_assignment(assignment)
     memberships.where(groupings: { assignment_id: assignment.id })
-               .includes(:grouping)
-               .count
+      .includes(:grouping)
+      .count
   end
 
   def get_groupings_by_assignment(assignment)
     groupings.where(assignment_id: assignment.id)
-             .includes(:students, :tas, :group, :assignment)
+      .includes(:students, :tas, :group, :assignment)
   end
 
   def get_membership_count_by_grade_entry_form(grade_entry_form)
     grade_entry_students.where('grade_entry_form_id = ?', grade_entry_form.id)
-                        .includes(:grade_entry_form)
-                        .count
+      .includes(:grade_entry_form)
+      .count
+  end
+
+  def grade_distribution_array(assignment, intervals = 20)
+    distribution = Array.new(intervals, 0)
+    assignment.groupings.joins(:tas)
+      .where(memberships: { user_id: id }).find_each do |grouping|
+      result = grouping.current_submission_used.get_latest_completed_result
+      next if result.nil?
+      distribution = update_distribution(distribution, result,
+                                         assignment.total_mark, intervals)
+    end # end of groupings loop
+    distribution.to_json
+  end
+
+  def update_distribution(distribution, result, out_of, intervals)
+    steps = 100 / intervals # number of percentage steps in each interval
+    percentage = (result.total_mark / out_of * 100).ceil
+    if percentage == 0 then distribution[0] += 1
```

Comment:
I was getting hound errors for the method being > 10 lines so I compacted the one-line bodies. That's also why I split up the `grade_distribution_array` method into two methods as well.